### PR TITLE
Add email template support

### PR DESCRIPTION
**Pull request by ayanko**
## Motivation

https://railsware.atlassian.net/browse/MT-5032

## Changes

* Introduce testing
* Add mailtrap client test
* Add email from template support
* Update README
* Define LTS nodejs version

## How to test

- [ ] Create email template
- [ ] Use example from README

## Images and GIFs

N/A

### CODE REVIEW

File: src/lib/types.ts [hunk 1]
```diff
@@ -1,14 +1,22 @@
...
+} & (MailContent | MailFromTemplateContent);
+
+type MailContent = {
+  subject: string;
+  category?: string;
+} & (TextMailContent | HTMLMailContent);
+
+type MailFromTemplateContent = {
+  template_uuid: string;
+  template_variables?: Record<string, string | number | boolean>;
```
Comment (leonid-shevtsov) [inline src/lib/types.ts:18]: can this be nested?

File: src/__tests__/mailtrap_client.test.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
+import { describe, expect, it, beforeAll, afterEach } from "@jest/globals";
```
Comment (leonid-shevtsov) [inline src/__tests__/mailtrap_client.test.ts:1]: nice! thanks for setting tests up

**Review by leonid-shevtsov — APPROVED** [review]

File: .eslintrc.js [hunk 3]
```diff
@@ -14,5 +14,5 @@ module.exports = {
   },
   plugins: ["@typescript-eslint"],
   rules: {},
-  ignorePatterns: ["dist/**", "examples/**", ".eslintrc.js"],
+  ignorePatterns: ["dist/**", "examples/**", "babel.config.js", "jest.config.js", ".eslintrc.js"],
```
Comment (ayanko) [inline .eslintrc.js:17]: Should/Can it be also TS files

File: src/__tests__/mailtrap_client.test.ts (hunk 2 above)
Comment (ayanko) [reply src/__tests__/mailtrap_client.test.ts:1]: Not sure which one is "kosher" way:

* `yarn add --dev @jest/globals`
* `yarn add --dev @types/jest`

https://jestjs.io/docs/getting-started#using-typescript

File: src/lib/types.ts (hunk 1 above)
Comment (ayanko) [reply src/lib/types.ts:18]: yes

**Comment by ayanko** [comment]
@VladimirTaytor ping

File: package.json [hunk 4]
```diff
@@ -20,15 +21,22 @@
     "axios": ">=0.27"
   },
   "devDependencies": {
+    "@babel/core": "^7.20.5",
+    "@babel/preset-env": "^7.20.2",
+    "@babel/preset-typescript": "^7.18.6",
+    "@jest/globals": "^29.3.1",
```
Comment (VladimirTaytor) [inline package.json:27]: During installation got such an error:
```
error @jest/globals@29.3.1: The engine "node" is incompatible with this module. Expected version "^14.15.0 || ^16.10.0 || >=18.0.0". Got "16.4.0"
```

Let's specify .node-version or .tool-versions (for asdf) for our repo
- I suggest using LTS right now it's 18.12.1

**Review by VladimirTaytor — COMMENTED** [review]
review in WIP need to switch device (out of electricity)

**Review by VladimirTaytor — APPROVED** [review]
Looks good 👍 

One comment according to PR description: 
- Let's don't expose our internal ticket ids in public repo, also Jira links 👀 
  - tho it's question to our Guild on how to maintain public repos 
  - not blocking this PR

File: .eslintrc.js (hunk 3 above)
Comment (ayanko) [reply .eslintrc.js:17]: Ok

File: package.json (hunk 4 above)
Comment (ayanko) [reply package.json:27]: Done.

**Pull request merged**